### PR TITLE
SNOW-1511108 SQLAlchemy heartbeat frequency parsed as string

### PR DESCRIPTION
**Pull request by sfc-gh-dszmolka**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

* SNOW-1511108
* https://github.com/snowflakedb/snowflake-sqlalchemy/issues/509

When coming from (snowflake-)sqlalchemy's Engine, the `client_session_keep_alive_heartbeat_frequency` option seems to come as string, which then will crash the connector with
```
..
  File "/usr/local/lib/python3.11/site-packages/snowflake/connector/connection.py", line 1680, in _validate_client_session_keep_alive_heartbeat_frequency
    elif self.client_session_keep_alive_heartbeat_frequency > real_max:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '>' not supported between instances of 'str' and 'int'
```

2. Fill out the following pre-review checklist:

- [ ] I am adding a new automated test(s) to verify correctness of my new code
- [ ] I am adding new logging messages
- [ ] I am adding a new telemetry message
- [ ] I am modifying authorization mechanisms
- [ ] I am adding new credentials
- [ ] I am modifying OCSP code
- [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

We already seem to have a protection mechanism against such occasions by converting the input to `int`, however when we do that, it's already too late because we already tried to compare `str` to `int` and failed with `TypeError`.

Functionality isn't changed, just moved the conversion to happen _before_ the comparison to `int` happens.

### CODE REVIEW

File: DESCRIPTION.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 # Release Notes
 
+- v3.x.x(TBD)
+  - ..
```
Comment (sfc-gh-mkeller) [inline DESCRIPTION.md:12]: Feel free to specify what you think is the next version going to be
```suggestion
- v3.11.1(TBD)
```

**Review by sfc-gh-mkeller — APPROVED** [review]
Could you please modify an already existing test case to check this?

:ship: :ship:

File: DESCRIPTION.md (hunk 1 above)
Comment (sfc-gh-dszmolka) [reply DESCRIPTION.md]: done, thank you !

**Comment by sfc-gh-dszmolka** [comment]
thank you again for the review @sfc-gh-mkeller ! could you please advise if skipping the 'Old Driver' test here would be okay? does it even have the change which we want to test here? 

(marked it as skipped it because it was the only one failed with the change, using PythonConnector 1.9.1 and all other tests/build passed)

if you approve with this approach, i'll merge. Thank you in advance !

**Pull request merged**